Framework/API/inc/MantidAPI/ISpectrum.h

```diff
@@ -169,6 +169,12 @@ class DLLExport ISpectrum {
   HistogramData::HistogramDx &mutableDx() {
     return mutableHistogramRef().mutableDx();
   }
+  HistogramData::HistogramY &mutableY() {
+    return mutableHistogramRef().mutableY();
+  }
+  HistogramData::HistogramE &mutableE() {
+    return mutableHistogramRef().mutableE();
+  }
```

Comment:
I think you need a throwing implementation of `mutableY` and `mutableE` in `EventList`. Otherwise `mutableHistogramRef()` gives a reference to the histogram in `EventList` that only contains `X`.